### PR TITLE
Fixed (No notification is given if an image is added. #1586) issue

### LINKED ISSUE #1586: No notification is given if an image is added.
**Describe the bug**
Under Add Event page, there is an Add Image option. When an image is added there is no notification or preview, so the user is unaware if an image was actually added. 

**To Reproduce**

Follow the steps outlined in the video below till 6:00
[Follow the steps outlined here till 6:00](https://www.youtube.com/watch?v=ix3eSReJj2k?t=325)

**Expected behavior**
Show a toast/popup of that the image was added and add a preview of said image in the post.

**Actual behavior**
There is no indication that an image is added. 

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Muhammed-EmadEldeen**
**What kind of change does this PR introduce?**
* When you try to add a new image when creating a post and after cropping it the image doesn't appear, So I solved the issue by getting the path of the cropped file first and then changing to File type.
* I added a snack bar informing the user that the image has been added successfully.

**Issue Number:**
Fixes #1586

**Did you add tests for your changes?**

Yes, I added tests for both add_post_model and multi_media_srevice.

**Snapshots/Videos:**
The Screen after uploading an image.
image

**If relevant, did you update the documentation?**

No, as this is a bug fix.

**Summary**
* Fixed the issue of the image not being visible when added and added a snackbar informing the user.
* https://github.com/PalisadoesFoundation/talawa/issues/1586

**Does this PR introduce a breaking change?**
No, but users can upload images now.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by Muhammed-EmadEldeen** [comment]
Are these changes sufficient or not @noman2002 ?

**Comment by noman2002** [comment]
@Muhammed-EmadEldeen Please fix the formatting.

**Comment by Muhammed-EmadEldeen** [comment]
> @Muhammed-EmadEldeen Please fix the formatting.

Done.
Do I need to create unit tests for multi_media_pick_service.dart or not?  As there is an open issue for making these tests.

**Comment by Muhammed-EmadEldeen** [comment]
Also, I found that issue no. #1681 is about the same bug which is solved here.
do I edit the pull request to say that it also closes that issue?

**Comment by palisadoes** [comment]
> Also, I found that issue no. #1681 is about the same bug which is solved here. do I edit the pull request to say that it also closes that issue?

That issue specifically speaks about errors when cropping the image. Aren't they different?

**Comment by palisadoes** [comment]
Please also fix the failing tests. Our test code coverage system will fail if either of these two conditions occur:

1) The overall code coverage drops below the target threshold of the repository
2) Any file in the pull request has code coverage levels below the repository threshold

We do this to ensure the overall reliability of the code base is maintained.

**Comment by palisadoes** [comment]
Your PR has been assigned reviewers. Please contact them either in this PR or via slack to get your code reviewed and approved. They should also be the:

1) arbitrators of future discussions with other contributors about the validity of your changes
2) point of contact for evaluating the validity of your work
3) person who verifies matching issues by others that should be closed.

Please make sure your code passes all tests.

**Comment by Muhammed-EmadEldeen** [comment]
> > Also, I found that issue no. #1681 is about the same bug which is solved here. do I edit the pull request to say that it also closes that issue?
> 
> That issue specifically speaks about errors when cropping the image. Aren't they different?

Issue #1586 is about no notification or preview showing when the user adds an image when I inspected the code I found that the code for previewing the image already exists but the image doesn't show. it turned out that when the user chooses an image the image cropper screen shows and when he crops the image it returns an error so the image doesn't appear in the preview as there is no image added. So I fixed that error which is the same error in issue #1681

**Comment by palisadoes** [comment]
Please proceed. Work with the assigned contributors to get your code reviewed and merged. I will not return to this PR until there is an approval.

**Comment by Muhammed-EmadEldeen** [comment]
I have completed tests for add_post_view_model file and the test for multi_medi_pick_service file already has an open issue so can you review my code @noman2002 @CyberWake @TheHazeEffect .Also as I mentioned above there is another issue which is #1681 which seems to be solved here , if that's so tell me so I can edit the PR to say that it solves that issue.

**Comment by palisadoes** [comment]
Please fix the failing tests. Our test code coverage system will fail if either of these two conditions occur:

1) The overall code coverage drops below the target threshold of the repository
2) Any file in the pull request has code coverage levels below the repository threshold

We do this to ensure the overall reliability of the code base is maintained. This is a prerequisite for getting your PR approved. Assigned reviewers regularly review the PR queue and tend to focus on PRs that are passing.

**Comment by palisadoes** [comment]
Please fix the failing tests. Our test code coverage system will fail if either of these two conditions occur:

1) The overall code coverage drops below the target threshold of the repository
2) Any file in the pull request has code coverage levels below the repository threshold

We do this to ensure the overall reliability of the code base is maintained. This is a prerequisite for getting your PR approved. Assigned reviewers regularly review the PR queue and tend to focus on PRs that are passing.

**Comment by Muhammed-EmadEldeen** [comment]
I finished the tests for the multi_media_pick_service, about the locator I don't know how can I test it or what to test in it.
can you review please @noman2002 @CyberWake @TheHazeEffect

**Comment by Muhammed-EmadEldeen** [comment]
I have completed the tests can you review @noman2002 @CyberWake @TheHazeEffect

File: test/helpers/test_helpers.mocks.dart [hunk 1]
```diff
@@ -3,51 +3,53 @@
...
+import 'package:connectivity_plus/connectivity_plus.dart' as _i26;
 import 'package:connectivity_plus_platform_interface/connectivity_plus_platform_interface.dart'
-    as _i25;
+    as _i27;
 import 'package:flutter/material.dart' as _i1;
 import 'package:graphql_flutter/graphql_flutter.dart' as _i3;
+import 'package:image_cropper/image_cropper.dart' as _i4;
+import 'package:image_picker/image_picker.dart' as _i13;
 import 'package:mockito/mockito.dart' as _i2;
-import 'package:qr_code_scanner/src/qr_code_scanner.dart' as _i30;
```
Comment (CyberWake) [inline test/helpers/test_helpers.mocks.dart:16]: Please try not changing so many things when not required.

Comment (Muhammed-EmadEldeen) [reply test/helpers/test_helpers.mocks.dart:16]: This file is generated by build_runner as I added mocks for two classes so I didn't change anything manually here

**Review by noman2002 — APPROVED** [review]

**Review by palisadoes — APPROVED** [review]

**Pull request merged**